src/main/java/net/hypixel/nerdbot/util/Util.java

```
@@ -168,20 +218,114 @@ public static File toFile(BufferedImage imageToSave) throws IOException {
+    @Deprecated
+    private static final String SURROUND_REGEX = "\\|([^|]+)\\||\\[([^\\[]+)\\]|\\{([^\\{]+)\\}|\\(([^\\(]+)\\)";
+
+    @Deprecated
+    public static Optional<String> getScuffedMinecraftIGN(Member member) {
+        // removes non-standard ascii characters from the discord nickname
+        String plainUsername = member.getEffectiveName().trim().replaceAll("[^\u0000-\u007F]", "");
+        String memberMCUsername = null;
+
+        // checks if the member's username has flair
+        if (!Pattern.matches(REGEX, plainUsername)) {
+            // removes start and end characters ([example], {example}, |example| or (example)).
+            // also strips spaces from the username
+            plainUsername = plainUsername.replaceAll(SURROUND_REGEX, "").replace(" ", "");
+            String[] splitUsername = plainUsername.split("[^a-zA-Z0-9_]");
+
+            // gets the first item that matches the name constraints
+            for (String item : splitUsername) {
+                if (Pattern.matches(REGEX, item)) {
+                    memberMCUsername = item;
+                    break;
+                }
+            }
+        } else {
+            memberMCUsername = plainUsername.replace(" ", "");
         }
 
-        Member sbnMember = guild.retrieveMemberById(user.getId()).complete();
-        if (sbnMember == null || sbnMember.getNickname() == null) {
-            return user.getEffectiveName();
+        return Optional.ofNullable(memberMCUsername);
+    }
+
+    public static MojangProfile getMojangProfile(String username) throws Exception {
+        try {
+            String url = String.format("https://api.mojang.com/users/profiles/minecraft/%s", username);
+            return NerdBotApp.GSON.fromJson(getHttpResponse(url).body(), MojangProfile.class);
+        } catch (Exception ex) {
+            throw new Exception("Unable to locate Minecraft UUID for `" + username + "`.", ex);
```

Would be better to throw a custom exception here instead